synapse/telepath.py

```diff
@@ -679,11 +679,16 @@ async def _teleLinkLoop(self):
                 logger.warning(f'telepath client ({url}): {e}')
                 await self.waitfini(timeout=self._t_conf.get('retrysleep', 0.2))
 
+    async def proxy(self, timeout=10):
+        await self.waitready(timeout=timeout)
+        return self._t_proxy
+
     async def _initTeleLink(self, url):
         if self._t_proxy is not None:
             await self._t_proxy.fini()
 
         self._t_proxy = await openurl(url, **self._t_opts)
+        self._t_methinfo = self._t_proxy.methinfo
```

Comment:
There is a potential bookkeeping issue with the Client; which is a reconnection should probably remove all existing attrs set with our __getattr__ hook in the event of an API change in the methinfo

Reply:
agreed, but lets do that in a subsequent delta ( we already have the problem now ).  I'll submit a bug.